# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsIOService.cpp
# Commit: 8baaed81dd29
# Full Hash: 8baaed81dd2949e0797546fa64ca3a09e14f7121
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-02-15 09:56:17
# Description:
#   Bug 1614730 - Make sure NSS is initialized before updating preference r=dragana
#   
#   It is possible that NSS is not initialized when receiving a preference update from parent process, so we have to call `EnsureNSSInitializedChromeOrContent()` before updating the preference.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D62555
# ==============================================================================

diff -r 61fceb15f2a5 -r 8baaed81dd29 netwerk/base/nsIOService.cpp
--- a/netwerk/base/nsIOService.cpp	Fri Feb 14 14:15:34 2020 +0000
+++ b/netwerk/base/nsIOService.cpp	Fri Feb 14 12:33:53 2020 +0000
@@ -315,6 +315,11 @@
 void nsIOService::OnTLSPrefChange(const char* aPref, void* aSelf) {
   MOZ_ASSERT(IsSocketProcessChild());
 
+  if (!EnsureNSSInitializedChromeOrContent()) {
+    LOG(("NSS not initialized."));
+    return;
+  }
+
   nsAutoCString pref(aPref);
   // The preferences listed in gCallbackSecurityPrefs need to be in sync with
   // the code in HandleTLSPrefChange() and SetValidationOptionsCommon().
